code/game/objects/items/explosives/mine.dm

```diff
@@ -345,53 +406,67 @@ taking that kind of thing into account, setting buffer_range = 0 or making them
 	if(duration || duration < 0)
 		extra_effects(victim)
 		if(duration > 0)
-			if(reusable)
+			if(CHECK_BITFIELD(mine_features, MINE_REUSABLE))
 				return deletion_timer = addtimer(CALLBACK(src, PROC_REF(disarm)), duration, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
 			return deletion_timer = addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(qdel), src), duration, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
 		return TRUE	//Don't stop
 
 	//Reusable mines do not delete themselves upon detonation, just go back to sleep
-	if(reusable)
+	if(CHECK_BITFIELD(mine_features, MINE_REUSABLE))
 		return disarm()
 
-	QDEL_LIST(triggers)
+	delete_detection_zone()
 	qdel(src)
 
 ///Shove any code for special effects caused by this mine here
 /obj/item/mine/proc/extra_effects(atom/movable/victim)
 	return
 
-///If this mine is volatile, explode! Easier to copy paste this into several places
-/obj/item/mine/proc/volatility_check()
-	if(volatile)
+///If this mine is volatile, explode! See top of mine.dm for volatility flags
+/obj/item/mine/proc/volatility_check(flag)
+	if(CHECK_BITFIELD(mine_features, flag))
 		//Let's make sure everyone knows it was not activated by normal circumstances
 		visible_message(span_danger("[icon2html(src, viewers(src))] \The [src]'s detonation mechanism is accidentally triggered!"))
 		trigger_explosion()
 
 //On explosion, mines trigger their own explosion, assuming they were not deleted straight away (larger explosions or probability)
-/obj/item/mine/ex_act()
+/obj/item/mine/ex_act(severity)
 	. = ..()
-	if(!QDELETED(src))
-		volatility_check()
+	if(QDELETED(src) || !armed)
+		return
 
-//Any EMP effects will render volatiles mines disabled
-/obj/item/mine/emp_act()
-	. = ..()
-	if(volatile)
-		disarm()
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			return
+		if(EXPLODE_HEAVY)
+			if(prob(25))
+				return
+		if(EXPLODE_LIGHT)
+			if(prob(50))
+				return
+		if(EXPLODE_WEAK)
+			return //not strong enough to detonate
+
+	volatility_check(MINE_VOLATILE_EXPLOSION)
```

Comment:
this is waht I had for AT mines specifically, why have you changed this to the base mine behavior?

Reply:
I liked it. This only matters if the mine has the flag to begin with. No reason to not reuse the code.